core/controllers/classifier.py

```diff
@@ -123,3 +129,61 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+def update_failed_jobs(job_models):
+    for job_model in job_models:
+        classifier_services.mark_training_job_failed(job_model.id)
+
+
+def fetch_training_data(exp_id, state):
```

Comment:
I guess it was decided that training_data be populated before VM picks up, so that VM gets the latest training_data. There is possibility that the training_data stored in job model may be changed by the time it is picked up. I am not sure if the job model is updated every time training_data changes. if it does happen then we don't need to do it in the controller.